temporal-sdk/src/main/java/io/temporal/internal/common/ProtobufTimeUtils.java

```diff
@@ -77,4 +131,12 @@ public static com.uber.m3.util.Duration toM3DurationSinceNow(Timestamp t) {
 
     return Timestamp.newBuilder().setSeconds(t.getEpochSecond()).setNanos(t.getNano()).build();
   }
+
+  @Nonnull
+  private static Duration roundToMillis(@Nonnull Duration exact) {
+    final long s = exact.getSeconds();
+    final int ns = exact.getNano();
+    final int ms = (ns + NANOS_PER_MILLI / 2) / NANOS_PER_MILLI;
```

Comment:
Couldn't we take the seconds and subtract them from `exact` and call `toMillis`? `toMillis` looks like it is doing a slightly different conversion. I worry about breaking any time skipping tests comparing exact timestamps
```
millis = Math.addExact(millis, tempNanos / NANOS_PER_MILLI);
```

Can we also add some tests for the conversion?